src/components/navbar/Navbar.tsx

```diff
@@ -1,30 +1,194 @@
+        type: FETCH_PROJECTS,
+        payload: response.data,
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  }, [getTokenSilently, projectsDispatch]);
 
-  return (
-    <div>
-      {!isAuthenticated && (
-        <button onClick={(): any => loginWithRedirect({})}>Log in</button>
-      )}
-
-      {isAuthenticated && (
-        <button onClick={(): any => logout()}>Log out</button>
-      )}
-
-      {isAuthenticated && (
-        <span>
-          <Link to="/">Home</Link>&nbsp;
-          <Link to="/profile">Profile</Link>
+  useEffect(() => {
+    fetchProjects();
+  }, [fetchProjects]);
+
+  const renderIcon = (): JSX.Element => {
+    return (
+      <Navbar.Brand>
+        <span className="navbar-icon">
+          <i className="fab fa-reddit" />
         </span>
-      )}
-    </div>
+      </Navbar.Brand>
+    );
+  };
+
+  const renderSearch = (): JSX.Element => {
+    return (
+      <Form inline className="navbar-search-form">
+        <InputGroup className="mb-3 navbar-search-input">
+          <FormControl placeholder="Search" />
+          <InputGroup.Append>
+            <Button variant="outline-secondary">
+              <i className="fas fa-search" />
+            </Button>
+          </InputGroup.Append>
+        </InputGroup>
+      </Form>
+    );
+  };
+
+  const renderProjectsDropdown = (): JSX.Element => {
+    return (
+      <Dropdown text="Projects" className="navbar-projects-dropdown" inline>
+        <Dropdown.Menu className="navbar-projects-dropdown-menu">
+          {isAuthenticated ? (
+            Object.keys(projectsState).length === 0 ? (
+              ""
+            ) : (
+              Object.keys(projectsState)
+                .slice(0, 5)
+                .map((project: any) => {
+                  return (
+                    <>
+                      <Dropdown.Item
+                        key={project.projectId}
+                        text={project.projectName}
+                      />
+                    </>
+                  );
+                })
+            )
+          ) : (
+            <>
+              <Dropdown.Item text="Project 1" />
+              <Dropdown.Item text="This is a really looooooooooooong project name" />
+              <Dropdown.Item text="Bug Tracker" />
+            </>
+          )}
+        </Dropdown.Menu>
+      </Dropdown>
+    );
+  };
+
+  const renderPlaceholderDropdown = (): JSX.Element => {
+    return (
+      <Dropdown
+        text="Placeholder"
+        className="navbar-placeholder-dropdown"
+        inline
+      >
+        <Dropdown.Menu>
+          <Dropdown.Item text="New" />
+        </Dropdown.Menu>
+      </Dropdown>
+    );
+  };
+
+  const renderSettings = (): JSX.Element => {
+    return (
+      <Dropdown
+        icon={<i className="fas fa-cog" />}
+        inline
+        className="navbar-settings"
+      >
+        <Dropdown.Menu fluid>
+          <Dropdown.Item text="Setting 1" />
```

Comment:
For all the dropdown items, you need to make sure they can redirect to another component/page.

Reply:
I can add onClick to these components when we have components/pages to redirect to. We don't have them at the moment.